Fix build failures with recent g++

src/BRepClass3d/BRepClass3d_SClassifier.cxx: In member function ‘void BRepClass3d_SClassifier::PerformInfinitePoint(BRepClass3d_SolidExplorer&, Standard_Real)’:
src/BRepClass3d/BRepClass3d_SClassifier.cxx:93:44: error: taking address of temporary [-fpermissive]

Index: oce/src/BRepClass3d/BRepClass3d_SClassifier.cxx
===================================================================
--- oce.orig/src/BRepClass3d/BRepClass3d_SClassifier.cxx
+++ oce/src/BRepClass3d/BRepClass3d_SClassifier.cxx
@@ -90,7 +90,8 @@ void BRepClass3d_SClassifier::PerformInf
   myState=2; 
   for(aSE.InitShell(); aSE.MoreShell() && nump<2;  aSE.NextShell()) { 
     for(aSE.InitFace();	aSE.MoreFace()  && nump<2; ) {
-      aF =*((TopoDS_Face*)&aSE.CurrentFace());
+      TopoDS_Shape aLocalShape = aSE.CurrentFace();
+      aF = TopoDS::Face(aLocalShape);
       aSE.NextFace();
       if(!nump) { 
 	nump++;
